demo/index.html

```diff
@@ -19,243 +19,107 @@
-          <paper-input name="password" label="Password" type="password" required auto-validate></paper-input>
-          <paper-checkbox name="read" required>You must check this box</paper-checkbox><br>
-          <paper-button raised onclick="_delayedSubmit(event)" disabled id="eventsDemoSubmit">
-            <paper-spinner id="spinner" hidden></paper-spinner>Submit</paper-button>
-          <paper-button raised onclick="_reset(event)">Reset</paper-button>
-          <div class="output"></div>
-        </form>
-        <script>
-          eventsDemo.addEventListener('change', function(event) {
-            // Validate the entire form to see if we should enable the `Submit` button.
-            eventsDemoSubmit.disabled = !eventsDemo.validate();
-          });
-          function _delayedSubmit(event) {
-            spinner.active = true;
-            spinner.hidden = false;
-            eventsDemoSubmit.disabled = true;
-            // Simulate a slow server response.
-            setTimeout(function() {
-              Polymer.dom(event).localTarget.parentElement.submit();
-            }, 1000);
-          }
-          function _reset(event) {
-            var form = Polymer.dom(event).localTarget.parentElement
-            form.reset();
-            form.querySelector('.output').innerHTML = '';
-          }
-          document.getElementById('eventsDemo').addEventListener('iron-form-submit', function(event) {
-            spinner.active = false;
-            spinner.hidden = true;
-            eventsDemoSubmit.disabled = false;
-            this.querySelector('.output').innerHTML = JSON.stringify(event.detail);
-          });
-        </script>
+            <paper-button raised onclick="submitHandler(event)">Submit</paper-button>
+            <paper-button raised onclick="resetHandler(event)">Reset</paper-button>
+          </form>
+        </iron-form>
       </template>
     </demo-snippet>
 
-    <h3>Example of using a native <code>form</code> to do a page redirect.</h3>
-
-    <p>Since an <code>iron-form</code> submits its elements with an XHR,
-    it cannot do a page redirect after submission. If you need one, you
-    can use a native <code>form</code> element and submit all the custom
-    elements using <code>hidden</code> inputs.</p>
+    <h3>Iron-form respects nested elements</h3>
     <demo-snippet>
       <template>
-        <!-- This form is only used to display the elements -->
-        <form is="iron-form" id="redirectDemo">
-          <paper-input name="name" label="Name" required></paper-input>
-          <paper-input name="password" label="Password" type="password" required></paper-input>
-          <paper-checkbox name="read" required>You must check this box</paper-checkbox><br>
-          <paper-button raised onclick="_nativeSubmit(event)">Submit</paper-button>
-        </form>
+        <iron-form>
+          <form method="get" action="http://localhost:3000/foo">
```

Comment:
oshit this is why the tests are failing on travis. genius, elliott, genius! (i don't remember what past monica did here)